apis/vmware/v1beta1/vspheremachine_types.go

```diff
@@ -34,7 +35,95 @@ type VSphereMachineVolume struct {
 	StorageClass string `json:"storageClass,omitempty"`
 }
 
+// RouteSpec defines a static route for a guest.
+type RouteSpec struct {
+	// To is either "default", or an IP4 address. IP6 is not supported yet.
+	//
+	// +kubebuilder:validation:Pattern=`^(default|([0-9]{1,3}\.){3}[0-9]{1,3}(\/[0-9]{1,2})?)$`
+	To string `json:"to"`
+
+	// Via is an IP4 address. IP6 is not supported yet.
+	//
+	// +kubebuilder:validation:Pattern=`^([0-9]{1,3}\.){3}[0-9]{1,3}$`
+	Via string `json:"via"`
+}
+
+// PartialObjectRef describes a reference to another object in the same
+// namespace as the referrer. The reference can be just a name but may also
+// include the referred resource's APIVersion and Kind.
+type PartialObjectRef struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Name refers to a unique resource in the current namespace.
+	Name string `json:"name"`
+}
+
+// InterfaceProperty defines properties of a network interface.
+type InterfaceProperty struct {
+	// Network is the name of the network resource to which this interface is
+	// connected.
+	Network PartialObjectRef `json:"network"`
+
+	// MTU is the Maximum Transmission Unit size in bytes.
+	//
+	// +kubebuilder:validation:Optional
+	MTU *int64 `json:"mtu,omitempty"`
+
+	// Routes is a list of optional, static routes.
+	//
+	// Please note this feature is available only with the following bootstrap
+	// providers: CloudInit.
+	//
+	// +kubebuilder:validation:Optional
+	Routes []RouteSpec `json:"routes,omitempty"`
+}
+
+// InterfaceSpec defines a network interface for a VSphereMachine.
+type InterfaceSpec struct {
```

Comment:
What about calling this SecondaryInterfaceSpec and then InterfaceProperty => InterfaceSpec?